iam/api-client/src/main/java/com/google/iam/snippets/ServiceAccountKeyDelete.java

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.iam.snippets;
+
+// [START iam_delete_key]
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.iam.v1.Iam;
+import com.google.api.services.iam.v1.IamScopes;
+import java.util.Collections;
+
+
+public class ServiceAccountKeyDelete {
+
+    public static void deleteKey(String fullKeyName) throws Exception {
+
+        GoogleCredential credential = GoogleCredential.getApplicationDefault()
```

Comment:
Since you are doing this section a lot, it's totally fine if you want to move it into a helper function after the main part of the snippet. This can make reading the snippet less tedious, since some of the boilerplate is removed:
```java
private void initService() throws IOException{
    // Use the Application Default Credentials strategy for authentication. For more info, please see:
   // https://cloud.google.com/docs/authentication/production#finding_credentials_automatically
    GoogleCredential credential = GoogleCredential.getApplicationDefault()
            .createScoped(Collections.singleton(IamScopes.CLOUD_PLATFORM));
    // Initialize the IAM service, which can be used to send requests to the IAM API.  
    Iam service = new Iam.Builder(
        GoogleNetHttpTransport.newTrustedTransport(), 
        JacksonFactory.getDefaultInstance(),
        credential
    ).setApplicationName("service-account-keys")
    .build();
}
```

Reply:
Done, let me know if I didn't do it properly